### DISCUSSION TITLE
StatelessWidget with Observer as a const?

### DISCUSSION THREAD

**Discussion by jshji252** [mobxjs/mobx.dart#999] [Q&A] [answered]
Hello team,

Thank you for amazing work, i hope i can contribute one day!

I was just thinking, when we have a Stateless widget (let's call it `Page()`) that uses a store and `Observer` widget, but does not use any arguments from the parent, Dart/Flutter allows us to instantiate that widget as a const, like `const Page()`. But Page() isn't technically `const`, is it? Or, is it const as its builder method's implementation doesn't change? I'd like to use `const` as much as possible to help performance. I don't understand all the technical details about `const` to find answers to this (but I am trying to!), so any help will be greatly appreciated!!

**Comment by amondnet** [comment, marked as answer]
Why does using const improve performance? Because the widget is not rebuilt on setState, i.e. the Page Widget itself is not rebuilt. Also, only the changed widgets inside the Page Widget are rebuilt by mobx, which improves performance.
